shop/src/main/java/ro/msg/learning/shop/domain/OrderDetail.java

```diff
@@ -0,0 +1,38 @@
+package ro.msg.learning.shop.domain;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.msg.learning.shop.domain.Keys.OrderDetailKey;
+
+import java.util.List;
+
+@Table(name = "orderdetail")
+@Entity
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class OrderDetail{
+    @EmbeddedId
+    private OrderDetailKey id;
+
+    @ManyToOne
+    @MapsId("order")
+    @JoinColumn(name = "order")
+    private Order order;
+
+    @ManyToOne
+    @MapsId("products")
+    @JoinColumn(name = "product")
+    private List<Product> products;
```

Comment:
An order detail should be defined for _only one_ product: for each product in an order, there is an order detail (this is why the order detail contains a composite key of an order ID and a product ID). So, you should not have a list of products here, but a single product: `private Product product;`

Have a look over the diagram again.